packages/visx-text/src/hooks/useText.ts

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState, useRef, useCallback } from 'react';
+import reduceCSSCalc from 'reduce-css-calc';
+import { usePreviousValue } from '@huse/previous-value';
+import { TextProps, WordsWithWidth, WordWithWidth } from '../types';
+import getStringWidth from '../util/getStringWidth';
+
+function isNumber(val: unknown): val is number {
+  return typeof val === 'number';
+}
+
+export function useText(props: TextProps): [WordsWithWidth[], string, string] {
+  const [wordsByLines, setWordsByLines] = useState<WordsWithWidth[]>([]);
+  const wordsWithWidth = useRef<WordWithWidth[]>([]);
+  const spaceWidth = useRef(0);
+  const prevProps = usePreviousValue(props);
+  const prevWordsByLines = usePreviousValue(wordsByLines);
+
+  const calculateWordsByLines = useCallback(
```

Comment:
Are those `useCallback` useless? Cause they all depend on `props`, which will not be reference equality at each `render`.